src/linking/remove_internal_symbols.cpp

```diff
@@ -149,7 +150,8 @@ void remove_internal_symbols(
     else if(is_function)
     {
       // body? not local (i.e., "static")?
-      if(has_body && !is_file_local)
+      if(has_body &&
+         (!is_file_local || (config.main==symbol.name.c_str())))
         get_symbols_rec(ns, symbol, exported);
```

Comment:
@kroening It's returning a `const char *` that we can compare to the `std::string` that is `config.main`. Without it, we get the following error message:

```
remove_internal_symbols.cpp: In function 'void remove_internal_symbols(symbol_tablet&)':
remove_internal_symbols.cpp:153:54: error: no match for 'operator==' (operand types are 'std::__cxx11::string {aka std::__cxx11::basic_string<char>}' and 'const irep_idt {aka const dstringt}')
       if(has_body || (!is_file_local && (config.main == symbol.name)))
```

If we need to change it, I can change the `c_str()` call to `as_string()`, but they appear to do exactly the same thing, with the difference that `c_str()` returns a C string whereas `as_string()` returns a c++ string.